src/DependencyInjection/HandlerCompilerPass.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcingBundle\DependencyInjection;
+
+use Patchlevel\EventSourcing\Attribute\Inject;
+use Patchlevel\EventSourcing\CommandBus\Handler\CreateAggregateHandler;
+use Patchlevel\EventSourcing\CommandBus\Handler\ServiceNotResolvable;
+use Patchlevel\EventSourcing\CommandBus\Handler\UpdateAggregateHandler;
+use Patchlevel\EventSourcing\CommandBus\HandlerFinder;
+use Patchlevel\EventSourcing\Metadata\AggregateRoot\AggregateRootRegistry;
+use Patchlevel\EventSourcing\Repository\RepositoryManager;
+use Patchlevel\EventSourcingBundle\CommandBus\SymfonyParameterResolver;
+use ReflectionAttribute;
+use ReflectionMethod;
+use Symfony\Component\DependencyInjection\Argument\ServiceLocatorArgument;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\TypeInfo\Type\ObjectType;
+use Symfony\Component\TypeInfo\TypeResolver\TypeResolver;
+
+use function sprintf;
+use function strtolower;
+
+/** @internal */
+final class HandlerCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->hasParameter('patchlevel_event_sourcing.aggregate_handlers.bus')) {
+            return;
+        }
+
+        $bus = $container->getParameter('patchlevel_event_sourcing.aggregate_handlers.bus');
+
+        /** @var AggregateRootRegistry $aggregateRootRegistry */
+        $aggregateRootRegistry = $container->get(AggregateRootRegistry::class);
+
+        foreach ($aggregateRootRegistry->aggregateClasses() as $aggregateName => $aggregateClass) {
+            $parameterResolverId = sprintf('.event_sourcing.handler_parameter_resolver.%s', $aggregateName);
+            $services = [];
+
+            foreach (HandlerFinder::findInClass($aggregateClass) as $aggregateHandler) {
+                $handlerId = strtolower(sprintf('event_sourcing.handler.%s.%s', $aggregateName, $aggregateHandler->method));
+                $handlerClass = $aggregateHandler->static ? CreateAggregateHandler::class : UpdateAggregateHandler::class;
+
+                $services += $this->services(new ReflectionMethod($aggregateClass, $aggregateHandler->method));
```

Comment:
```suggestion
                $services = [...$services, ...$this->services(new ReflectionMethod($aggregateClass, $aggregateHandler->method))];
```

otherwise we could run again into problems replacing entries in `$service`, couldn't we?

Reply:
Ah, nvm - we are using a key here built with `methodName` + `paramName`, this should be unique.